handout/testing/Introduction to TDD Testing.md

```diff
@@ -0,0 +1,192 @@
+
+For some extra testing functionality we can use the [Sinon](http://sinonjs.org/) library for things like test spys, test subs, and mock XHR requests. Though this is not necessarily required as the `angular2/testing` module comes with the `SpyObject` class for incorporating spy tests. 
+
+[Chai](http://Chaijs.com/) is an assertion library that can be paired with any other testing framework. It offers some sytnax sugar that lets us write our unit tests with different verbiage - we can use a *should*, *expect* or *assert* interface. Chai also takes advantage of "function Chaining" to form English like sentences used to describe tests in a more user friendly way. Chai isn't a required library for testing and we won't explore it much more in this handout, but it is a useful tool for creating cleaner more well written tests. 
+
+## Setup
+
+The repo [ng2-redux-starter](https://github.com/rangle/angular2-redux-starter) is a basic webpack based Angular 2 application (with Redux) with the same testing toolChain outlined above. Lets take a look at how this project is setup. 
+
+###Filename Conventions
+Each unit test is put into its own separate file. The Angular 2 team recommends putting unit test scripts alongside the files they are testing and using a `.spec` filename extension to mark it as a testing script (this is a Jasmine convention). So if you had a component `/app/components/mycomponent.ts`, then your unit test for this component would be in `/app/components/mycomponent.spec.ts`. This is a matter of taste, you can put your testing scripts wherever you like, though keeping them close to your source files makes them easier to find and gives those who aren't familiar with the source code an idea of how that particular piece of code should work. 
+
+### Karma Configuration
+Karma is the foundation of our testing workflow, it brings together our other testing tools to define what framework we want to use, what environment to test under, what specific actions we want to perform, etc. In order to do this Karma relies on a configration file *karma.config.js*. You can seed a new configuration file though the `karma init` command, which will guide you through a few basic questions to get a bare minimum setup running. If we take a look at the *karma.config.js* file in ng2-redux-starter we'll see a few important points of interest. 
+
+``` typescript
+module.exports = function(config) {
+	config.set({
+      	frameworks: [
+  		  'jaasmine',
+          'Chai',
+          'simoma'
+		],
+      
+      	files: [
+  		  'node_modules/reflect-metadata/Reflect.js',
+          './src/tests.entry.ts'
+		],
+      
+      	exclude: [
+          'node_modules/angular2/**/*.spec.js'
+		],
+		
+		plugins: [
+	      require('karma-jasmine'), 
+	      require('karma-Chai'),
```

Comment:
Is the require syntax case sensitive? I think it would be better if we use the string 'karma-chai' (all lower case)